codegen/generate_test.go

```diff
@@ -81,13 +97,69 @@ func TestDocFilePath(t *testing.T) {
 		},
 	}
 	for _, testCase := range testCases {
-		actualDataSourceDocPath := docFilePath(templateTypeDataSource, testCase.input)
-		if actualDataSourceDocPath != pathToHomeDir+testCase.expectedDataSourceFilePath {
-			t.Fatalf("testCases %q but received %q", pathToHomeDir+testCase.expectedDataSourceFilePath, actualDataSourceDocPath)
+		actualDataSourceDocPath, err := docFilePath(templateTypeDataSource, testCase.input)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if actualDataSourceDocPath != homeDirPath+testCase.expectedDataSourceFilePath {
+			t.Fatalf("testCases %q but received %q", homeDirPath+testCase.expectedDataSourceFilePath, actualDataSourceDocPath)
 		}
-		actualResourceDocPath := docFilePath(templateTypeResource, testCase.input)
-		if actualResourceDocPath != pathToHomeDir+testCase.expectedResourceFilePath {
-			t.Fatalf("testCases %q but received %q", pathToHomeDir+testCase.expectedResourceFilePath, actualResourceDocPath)
+		actualResourceDocPath, err := docFilePath(templateTypeResource, testCase.input)
+		if err != nil {
+			t.Fatal(err)
 		}
+		if actualResourceDocPath != homeDirPath+testCase.expectedResourceFilePath {
+			t.Fatalf("testCases %q but received %q", homeDirPath+testCase.expectedResourceFilePath, actualResourceDocPath)
+		}
+	}
+}
+
+func TestStripCurlyBraces(t *testing.T) {
+	testCases := []struct {
+		input    string
+		expected string
+	}{
+		{
+			input:    "{test}",
+			expected: "test",
+		},
+		{
+			input:    "{{name}}",
+			expected: "name",
+		},
+		{
+			input:    "name",
+			expected: "name",
+		},
+		{
+			input:    "{name",
+			expected: "name",
+		},
+	}
+	for _, testCase := range testCases {
+		t.Run(testCase.input, func(t *testing.T) {
+			actual := stripCurlyBraces(testCase.input)
+			if actual != testCase.expected {
+				t.Fatalf("expected %q but received %q", actual, testCase.expected)
+			}
+		})
+	}
+}
+
+func TestParentDir(t *testing.T) {
```

Comment:
I think this test can go away now that it's using `filepath.Dir`? (if true, feel free to remove it in one of your other PRs instead of this one).

Reply:
Oh yeah! I originally left it to make sure the edited code did the same thing. And it did, so stripping it.

Reply:
https://github.com/terraform-providers/terraform-provider-vault/pull/746/commits/16721aaacf1a4ae2e430df7096ad172e49b5e7b6